Don't Require Keys To Be File Backed

spjmurray opened the pull request:
Inspired by pull request #8 this keys keys on the client name and allows the
key to be optionally generated at a specified path, also handling all the CI
jobs that were failing :)  Seeing as this is a fairly major/breaking change
I've taken the liberty of cleaning up how capabilities are handled i.e. type
checking handles keys and key generation becomes a simple loop.

jordanconway commented:
@brwyatt 4704db9 Seems to break on CentOS 7 as lsblk with the parttype field is only in util-linux >= 2.25 and CentOS 7.2 currently only has version 2.23.2

brwyatt commented:
@mricon @jordanconway Nice catch!

I'm going to try and spend some time to get this taken care of, shouldn't be too difficult.

brwyatt commented:
@mricon @jordanconway Let me know if the latest commit works for y'all!

jordanconway commented:
@brwyatt It looks good for this change - although there seem to be some issues with the prerequisite for redhat-lsb-core. You're trying to access $::lsbmajdistrelease in params.pp but it's provided by Package redhat-lsb-core which is only installed in install.pp
It works perfectly if  redhat-lsb-core is already installed though.

brwyatt commented:
@jordanconway Glad it worked!

As to the $::lsbmajdistrelease, though... unfortunately, that's not something I've added, as it was part of the module before I started using it. A lot of modules use that fact, so it might be good to have redhat-lsb-core installed in base images. However I think there are also some alternate facts as well, so I might be able to put in another PR to change to one of those, but it would be pretty low on my priority list (most systems I work with have the LSB packages installed by default), but the change should be trivial.

jordanconway commented:
@brwyatt I didn't see it in the datacenterd params.pp https://github.com/datacentred/puppet-ceph/blob/master/manifests/params.pp so I thought it was something in one of your PRs.
The issue isn't that it's being used, it's that it's being used before install.pp is called to install it.

brwyatt commented:
@jordanconway You're right, my mistake! I forgot about that one, it is part of PR #4.

I'll comment over there about this.

Commit:
Don't Require Keys To Be File Backed

Inspired by pull request #8 this keys keys on the client name and allows the
key to be optionally generated at a specified path, also handling all the CI
jobs that were failing :)  Seeing as this is a fairly major/breaking change
I've taken the liberty of cleaning up how capabilities are handled i.e. type
checking handles keys and key generation becomes a simple loop.

spjmurray commented:
Breaking API change, semver to 4.0.0 for release

brwyatt commented:
I believe commit 66787901 resolves the original issue by switching to `sgdisk`.
